benchmark/Math/NumberTheory/PrimitiveRootsBench.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns #-}
+
+module Math.NumberTheory.PrimitiveRootsBench
+  ( benchSuite
+  ) where
+
+import Gauge.Main
+import Data.Maybe
+
+import Math.NumberTheory.Moduli.PrimitiveRoot
+import Math.NumberTheory.UniqueFactorisation
+
+primRootWrap :: Integer -> Word -> Integer -> Bool
+primRootWrap p k g = isPrimitiveRoot' (CGOddPrimePower p' k) g
+  where p' = fromJust $ isPrime p
+
+primRootWrap2 :: Integer -> Word -> Integer -> Bool
+primRootWrap2 p k g = isPrimitiveRoot' (CGDoubleOddPrimePower p' k) g
+  where p' = fromJust $ isPrime p
+
+-- contrived function which forces the cyclic group to be fully calculated 
+forceCyclic :: Maybe (CyclicGroup a) -> Int
+forceCyclic Nothing    = 0
+forceCyclic (Just CG2) = -2
+forceCyclic (Just CG4) = -1
+forceCyclic (Just (CGOddPrimePower !_p !_k)) = 1
+forceCyclic (Just (CGDoubleOddPrimePower !_p !_k)) = 2
```

Comment:
Let us rather derive `NFData` instance for `CyclicGroup`, similar to benchmarks of Gaussian integers: [one](https://github.com/cartazio/arithmoi/blob/master/Math/NumberTheory/GaussianIntegers.hs#L55) and [two](https://github.com/cartazio/arithmoi/blob/master/benchmark/Math/NumberTheory/GaussianIntegersBench.hs#L14).

Reply:
Fixed.